pkg/s3select/csv-input.go

```diff
@@ -0,0 +1,224 @@
+		} else {
+			for i, value := range dec {
+				record["_"+strconv.Itoa(i)] = value
+			}
+		}
+		return record, nil
+	}
+	return nil, nil
+}
+
+// It returns the delimitter specified in input request
+func (reader *CSVInput) getOutputFieldDelimiter() string {
+	return reader.options.OutputFieldDelimiter
+}
+
+// Returns true or false depending upon the header of CSV
+func (reader *CSVInput) hasHeader() bool {
+	return reader.options.HasHeader
+}
+
+// Return the header for CSV
+func (reader *CSVInput) getHeader() []string {
+	return reader.header
+}
+
+// Return the Select Expression for CSV
+func (reader *CSVInput) getExpression() string {
+	return reader.options.Expression
+}
+
+// Updates the Bytes returned for CSV
+func (reader *CSVInput) bytesReturned(size int64) {
+	reader.stats.BytesReturned += size
+}
+
+// Header returns the header of the reader. Either the first row if a header
+// set in the options, or c#, where # is the column number, starting with 0.
+func (reader *CSVInput) Header() []string {
+	return reader.header
+}
+
+// ReadRecord reads a single record from the . Always returns successfully.
+// If the record is empty, an empty []string is returned.
+// Record expand to match the current row size, adding blank fields as needed.
+// Records never return less then the number of fields in the first row.
+// Returns nil on EOF
+// In the event of a parse error due to an invalid record, it is logged, and
+// an empty []string is returned with the number of fields in the first row,
+// as if the record were empty.
+//
+// In general, this is a very tolerant of problems  reader.
+func (reader *CSVInput) ReadRecord() []string {
+	var row []string
+	var fileErr error
+
+	if reader.firstRow != nil {
+		row = reader.firstRow
+		reader.firstRow = nil
+		return row
+	}
+
+	row, fileErr = reader.reader.Read()
+	emptysToAppend := reader.minOutputLength - len(row)
+	if fileErr == io.EOF || fileErr == io.ErrClosedPipe {
+		return nil
+	} else if _, ok := fileErr.(*csv.ParseError); ok {
+		emptysToAppend = reader.minOutputLength
+	}
+
+	if emptysToAppend > 0 {
+		for counter := 0; counter < emptysToAppend; counter++ {
+			row = append(row, "")
+		}
+	}
+
+	return row
+}
+
+// createStatXML is the function which does the marshaling from the stat
+// structs into XML so that the progress and stat message can be sent
+func (reader *CSVInput) createStatXML() (string, error) {
+	if reader.options.Compressed == "NONE" {
+		reader.stats.BytesProcessed = reader.options.StreamSize
+		reader.stats.BytesScanned = reader.stats.BytesProcessed
+	}
+	out, err := xml.Marshal(&stats{
+		BytesScanned:   reader.stats.BytesScanned,
+		BytesProcessed: reader.stats.BytesProcessed,
+		BytesReturned:  reader.stats.BytesReturned,
+	})
+	if err != nil {
+		return "", err
+	}
+	return xml.Header + string(out), nil
+}
+
+// createProgressXML is the function which does the marshaling from the progress structs into XML so that the progress and stat message can be sent
```

Comment:
can this comment be written in two lines?

Reply:
Done